common/beerocks/bwl/nl80211/sta_wlan_hal_nl80211.cpp

```diff
@@ -69,7 +227,161 @@ std::string sta_wlan_hal_nl80211::get_bssid() { return m_active_bssid; }
 
 bool sta_wlan_hal_nl80211::process_nl80211_event(parsed_obj_map_t &parsed_obj) { return true; }
 
-bool sta_wlan_hal_nl80211::update_status() { return false; }
+bool sta_wlan_hal_nl80211::update_status()
+{
+    ConnectionStatus connection_status;
+    if (!read_status(connection_status)) {
+        LOG(ERROR) << "Failed reading connection status for iface: " << get_iface_name();
+        return false;
+    }
+    update_status(connection_status);
+
+    return true;
+}
+
+// Add a new network and return the ID
+int sta_wlan_hal_nl80211::add_network()
+{
+    char *reply = nullptr;
+
+    std::string cmd = "ADD_NETWORK";
+
+    // Send command
+    if (!wpa_ctrl_send_msg(cmd, &reply)) {
+        LOG(ERROR) << "ADD_NETWORK failed!";
+        return false;
+    }
+
+    // Return the newly added network ID
+    return beerocks::string_utils::stoi(reply);
+}
+
+bool sta_wlan_hal_nl80211::set_network(int network_id, const std::string &param,
+                                       const std::string &value)
+{
+    const std::string cmd = "SET_NETWORK " + std::to_string(network_id) + " " + param + " " + value;
+    if (!wpa_ctrl_send_msg(cmd)) {
+        LOG(ERROR) << "SET_NETWORK failed!";
+        return false;
+    }
+    return true;
+}
+
+// Update network parameters
+bool sta_wlan_hal_nl80211::set_network_params(int network_id, const std::string &ssid,
+                                              const std::string &bssid, WiFiSec sec,
+                                              bool mem_only_psk, const std::string &pass,
+                                              bool hidden_ssid, int freq)
+{
+    // Set SSID
+    if (!set_network(network_id, "ssid", "\"" + ssid + "\"")) {
+        LOG(ERROR) << "set_network_params ssid failed!";
+        return false;
+    }
+
+    // Set BSSID
+    if (!set_network(network_id, "bssid", bssid)) {
+        LOG(ERROR) << "set_network_params bssid failed!";
+        return false;
+    }
+
+    // optional: set frequency
+    if ((freq > 0) && (!set_network(network_id, "freq_list", std::to_string(freq)))) {
+        LOG(ERROR) << "Failed setting frequency on iface " << get_iface_name();
+        return false;
+    }
+
+    // Security and Password
+    if (!set_network(network_id, "key_mgmt", nl80211_security_val(sec))) {
+        LOG(ERROR) << "set_network_params key_mgmt failed!";
+        return false;
+    }
+
+    if (hidden_ssid && (!set_network(network_id, "scan_ssid", "1"))) {
+        LOG(ERROR) << "set_network_params hidden failed!";
+        return false;
+    }
+
+    if (mem_only_psk) {
+        if (!set_network(network_id, "mem_only_psk", "1")) {
+            LOG(ERROR) << "set_network_params hidden failed!";
+            return false;
+        }
```

Comment:
```suggestion
    if (mem_only_psk) {
        if (!set_network(network_id, "mem_only_psk", "1")) {
            LOG(ERROR) << "set_network_params memory-only-psk failed!";
            return false;
        }
```
and I think you should provide some info about what mem_only_psk is